### PR TITLE
Add an error example

### PR DESCRIPTION
**Pull request by rexagod**
I didn't find the current "error" section to be convincing in it's expression and semantics, hence I added this so anyone can quickly get a clear idea about it.

PS. Thank you for this cheat sheet!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -582,6 +582,32 @@ func main() {
...
+```go
+type number struct {
+	n int
+}
+
+// define a sqrt function on "number" type that returns
+// a float64 and error value based on the input
+func (N number) sqrt() (val float64, err error) {
+	if N.n < 0 {
+		err = errors.New("Negative value!")
```
Comment (a8m) [inline README.md:596]:
```suggestion
		err = errors.New("negative value")
```
Error strings should not be capitalized or end with punctuation. See https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: README.md [hunk 2]
```diff
@@ -582,6 +582,32 @@ func main() {
...
+```
+Here's an example:
+```go
+type number struct {
+	n int
+}
+
+// define a sqrt function on "number" type that returns
+// a float64 and error value based on the input
+func (N number) sqrt() (val float64, err error) {
```
Comment (a8m) [inline README.md:594]: Please merge this example with the one above by replacing the `doStuff` function.

Also, this struct is not really for this example. Let's define it only as a single function.

Thanks

**Review by a8m — COMMENTED** [review]
Thanks for your contribution @rexagod

File: README.md [hunk 3]
```diff
@@ -582,6 +582,32 @@ func main() {
...
+		// err == nil here
+		val = math.Sqrt(float64(N.n))
+	}
+	return
+}
+
+func main() {
+	N := number{-1}
+	_, err := N.sqrt()
+	fmt.Print(err == nil) // false
```
Comment (codingsince1985) [inline README.md:608]: on top of what @a8m suggested, it'd also be better to demo `err.Error()` as well.
```suggestion
	fmt.Print(err == nil) // false
	fmt.Print(err.Error()) // negative value
```

**Pull request closed without merging**